p4_symbolic/bmv2/test.cc

```diff
@@ -43,34 +36,36 @@ void WriteFile(char path[], const std::string& content) {
 }
 
 // The main test routine for parsing bmv2 json with protobuf.
-// Parses bmv2 json that is fed in through stdin and dumps
-// the resulting native protobuf and json data to files.
-// Expects the paths of the protobuf output file and json
-// output file to be passed as command line arguments respectively.
+// Parses bmv2 json file and dumps the resulting bmv2 protobuf
+// and json data to files.
+// Expects the paths of the input json file and the protobuf and json output
+// files to be passed as command line arguments in order.
 int main(int argc, char* argv[]) {
   // Verify link and compile versions are the same.
   GOOGLE_PROTOBUF_VERIFY_VERSION;
 
   // Validate command line arguments.
-  if (argc != 3) {
-    std::cout << "Usage: ./main <protobuf output file> <json output file>."
+  if (argc != 4) {
+    std::cout << "Usage: ./main <input JSON file> <protobuf output file> <json "
+                 "output file>."
               << std::endl;
     return 0;
   }
 
-  // Read input json from stdin.
-  std::string input = ReadStdin();
-
-  // Parsing JSON with protobuf.
-  p4_symbolic::bmv2::P4Program p4_buf;
-  google::protobuf::util::JsonParseOptions parsing_options;
-  parsing_options.ignore_unknown_fields = true;
-  google::protobuf::util::JsonStringToMessage(input, &p4_buf, parsing_options);
+  // Parse JSON using bmv2.cc.
+  std::string input(argv[1]);
+  pdpi::StatusOr<p4_symbolic::bmv2::P4Program> status =
+      p4_symbolic::bmv2::ParseBmv2Json(input);
+  if (!status.ok()) {
+    std::cerr << "Error reading input file: " << status.status() << std::endl;
+    return 1;
+  }
 
   // Dumping protobuf.
   std::string protobuf_output_str;
-  google::protobuf::TextFormat::PrintToString(p4_buf, &protobuf_output_str);
-  WriteFile(argv[1], protobuf_output_str);
+  google::protobuf::TextFormat::PrintToString(status.value(),
+                                              &protobuf_output_str);
+  WriteFile(argv[2], protobuf_output_str);
```

Comment:
Use `status.value().DebugString()`?

Reply:
I tested both and they seem to produce the same output, but I prefer this because I think it expresses the intent better. This is not meant to be a debugging or human readable thing (even though it is human readable). But it is meant to be a standard deterministic format that can be parsed and that is consistent across versions etc for golden file testing.

Reply:
That's what DebugString is :)

DebugString is used very widely for this sort of thing at Google (we have 3 million calls of DebugString vs 170k of PrintToString).

But up to you.

Reply:
ok cool. will do that then.